packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -61,39 +89,153 @@ contract FeeHandler is Ownable, Initializable, UsingRegistry, ICeloVersionedCont
+    require(tokens.length == newMaxSlippages.length, "maxSlippage length should match tokens");
 
     _transferOwnership(msg.sender);
     setRegistry(_registryAddress);
+    _setFeeBeneficiary(newFeeBeneficiary);
 
     for (uint256 i = 0; i < tokens.length; i++) {
+      _addToken(tokens[i], handlers[i]);
       _setDailyBurnLimit(tokens[i], newLimits[i]);
       _setMaxSplippage(tokens[i], newMaxSlippages[i]);
-      // Mento tokens don't need to set a router
-      if (newRouters[i] != address(0)) {
-        _setRouter(tokens[i], newRouters[i]);
-      }
     }
   }
 
+  function getTokenHandler(address tokenAddress) external view returns (address) {
+    return tokenStates[tokenAddress].handler;
+  }
+
+  function getTokenActive(address tokenAddress) external view returns (bool) {
+    return tokenStates[tokenAddress].active;
+  }
+
+  function getTokenMaxSlippage(address tokenAddress) external view returns (uint256) {
+    return FixidityLib.unwrap(tokenStates[tokenAddress].maxSlippage);
+  }
+
+  function getTokenDailyBurnLimit(address tokenAddress) external view returns (uint256) {
+    return tokenStates[tokenAddress].dailyBurnLimit;
+  }
+
+  function getTokenCurrentDayLimit(address tokenAddress) external view returns (uint256) {
+    return tokenStates[tokenAddress].currentDayLimit;
+  }
+
+  function getTokenToDistribute(address tokenAddress) external view returns (uint256) {
+    return tokenStates[tokenAddress].toDistribute;
+  }
+
+  function setFeeBeneficiary(address beneficiary) external onlyOwner {
+    return _setFeeBeneficiary(beneficiary);
+  }
+
+  function _setFeeBeneficiary(address beneficiary) private {
+    feeBeneficiary = beneficiary;
+    emit FeeBeneficiarySet(beneficiary);
+  }
+
+  function _setBurnFraction(uint256 newFraction) private {
+    FixidityLib.Fraction memory fraction = FixidityLib.wrap(newFraction);
+    require(
+      FixidityLib.lte(fraction, FixidityLib.fixed1()),
+      "Burn fraction must be less than or equal to 1"
+    );
+    burnFraction = fraction;
+    emit BurnFractionSet(newFraction);
+  }
+
+  function setBurnFraction(uint256 fraction) external onlyOwner {
+    return _setBurnFraction(fraction);
+  }
+
+  function sell(address tokenAddress) external {
+    return _sell(tokenAddress);
+  }
+
+  function _addToken(address tokenAddress, address handlerAddress) private {
+    IFeeHandlerSeller(handlerAddress);
+
+    // Check that the contract implements the interface
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    tokenState.active = true;
+    tokenState.handler = handlerAddress;
+
+    activeTokens.push(tokenAddress);
+  }
+
+  function addToken(address tokenAddress, address handlerAddress) external onlyOwner {
+    _addToken(tokenAddress, handlerAddress);
+  }
+
+  function getActiveTokens() public view returns (address[] memory) {
+    return activeTokens;
+  }
+
+  function removeToken(address tokenAddress) external {}
```

Comment:
Can we implement removeToken method? Since currently we can only add new tokens

Reply:
This is for the ultragreen branch, I'll implement the rest after